### PR TITLE
Hyperopt params-as-seed

### PR DESCRIPTION
**Pull request by laloch**
### Description

This PR adds a new feature to Hyperopt that allows users to "seed" the optimization process with their strategy's current parameters.

Currently, Hyperopt starts with completely random points. This feature allows users who have a working strategy to start the search from their known "good" configuration, potentially speeding up the discovery of improvements or fine-tuning existing parameters.

### Usage

```bash
freqtrade hyperopt --params-as-seed --spaces buy sell ...
```

### Implementation Details

*   **Flag:** Added `--params-as-seed` boolean flag.
*   **Fully Seeded:** If only standard parameter spaces (buy, sell, stoploss, trailing, max_open_trades) are optimized, a single trial is enqueued with the current values.
*   **Partially Seeded (ROI handling):** If the optimization includes spaces that cannot be reconstructed from the strategy file (specifically `roi`), the system automatically enqueues **5 trials**. These trials keep the known parameters (e.g., `buy`/`sell`) fixed while testing 5 different random ROI tables. This helps establish a strong baseline even when the ROI table is being re-optimized.

### Disclaimer

This code is partially AI-generated.

### CODE REVIEW

**Comment by froggleston** [comment]
Hyperopting already supports loading initial parameters as starting points using the `load` option: https://www.freqtrade.io/en/stable/hyperopt/#parameter-options

File: freqtrade/optimize/hyperopt/hyperopt.py [hunk 1]
```diff
@@ -228,6 +228,77 @@ def start(self) -> None:
...
+                # 1. Strategy Parameters (buy/sell/protection/etc)
+                for attr_name, attr in strat.enumerate_parameters():
+                    if attr.optimize and attr.in_space:
+                        current_params[attr_name] = attr.value
+
+                # 2. Stoploss
+                if "stoploss" in self.hyperopter.o_dimensions:
+                    current_params["stoploss"] = strat.stoploss
+
+                # 3. Max Open Trades
```
Comment (xmatthias) [inline freqtrade/optimize/hyperopt/hyperopt.py:246]: we won't need these numbers - they're a sign of non-reviewed AI code, really.

File: freqtrade/optimize/hyperopt/hyperopt.py [hunk 2]
```diff
@@ -228,6 +228,77 @@ def start(self) -> None:
         logger.info(f"Number of parallel jobs set as: {config_jobs}")
 
         self.opt = self.hyperopter.get_optimizer(self.random_state)
+
+        if self.config.get("hyperopt_params_as_seed", False):
+            try:
```
Comment (xmatthias) [inline freqtrade/optimize/hyperopt/hyperopt.py:233]: that whole section should be in it's own function (maybe only the `current_params` generation ...)

it should also have tests - as that's non-trivial code that's pretty easy to break accidentally.

**Review by xmatthias — CHANGES_REQUESTED** [review]
I think the initial premise is nice on first thought - though i'm not sure it holds on 2nd thought 😆 

If i understood this right (in a simplified sample with only one dimension): 
If my existing strategy used `rsi = 35` as value - it'll start from there and optimize from there (so exageratingly - the next point would be rsi=36 / rsi=34).

In current behavior - we'd start with a totally random RSI value - which _could_ be 35 - but also 55 or anything else within the search space.

---

This sounds good in theory (to the point i'm not sure the logic shouldn't be reversed, making this the default)  - but will in reality hinder reproducibility (prior start-value was 35 - new startvalue from the strategy is 36 - so `--random-state` is no longer guaranteeing the same starting point).

I'm not entirely sure if you'll actually gain much from it, or if you'll instead be losing more as you'll start from your biased local minima - missing out on much bigger local minima's in other spaces (not a problem in 1d space - but still a problem of hyperopt in general).

---

Especially the ROI part is a point of concern (assuming it's correct what you state - which i'm not entirely sure about - as the logic for ROI should be reversible just as you did for trailing-stop - though it might be out of reach for an LLM - so you'll need to put human thought into it 😆 ).

you'll also need to be quite careful with custom space definitions.
stoploss / roi / max_open_trades, (....) spaces can be [overridden](https://www.freqtrade.io/en/latest/advanced-hyperopt/#overriding-pre-defined-spaces) - where your assumptions won't hold true any longer (can be a point to add as warning to the docs just as well).

---

Current code will need some work (at first glance, it looks (at least partially) like AI output without much review (At least half of the comments have no meaning if you read the next line) - though it's also much less "sloppy" than what we're used to - so i'll remove the AI Slop label - as that's intended for really bad slop ... 

It'll also definitely need tests.

**Comment by laloch** [comment]
> Hyperopting already supports loading initial parameters as starting points using the `load` option: https://www.freqtrade.io/en/stable/hyperopt/#parameter-options

That's completely different thing.

**Comment by laloch** [comment]
Hey @xmatthias, thanks for your time. You're right - much of the code is generated by Gemini 3 Pro. It's by far the best agent I've seen so far. However, I had to first figure out if the idea is viable at all and how to implement it. This initial pull request was mainly to get some feedback. So, thank you very much!

> If i understood this right (in a simplified sample with only one dimension): If my existing strategy used `rsi = 35` as value - it'll start from there and optimize from there (so exageratingly - the next point would be rsi=36 / rsi=34).

That's right.

> This sounds good in theory (to the point i'm not sure the logic shouldn't be reversed, making this the default) - but will in reality hinder reproducibility (prior start-value was 35 - new startvalue from the strategy is 36 - so `--random-state` is no longer guaranteeing the same starting point).

The functionality should rather be complementary to the `--random-state`.

> I'm not entirely sure if you'll actually gain much from it, or if you'll instead be losing more as you'll start from your biased local minima - missing out on much bigger local minima's in other spaces (not a problem in 1d space - but still a problem of hyperopt in general).

There are three scenarios I wanted this for:
- Continuation of interrupted, or too short hyperopt run (not enough epochs) with promising results.
- I want to keep the best/selected result and switch the loss function.
- Attempt to improve on manually tuned parameters.

> Especially the ROI part is a point of concern (assuming it's correct what you state - which i'm not entirely sure about - as the logic for ROI should be reversible just as you did for trailing-stop - though it might be out of reach for an LLM - so you'll need to put human thought into it 😆 ).

I'll try to come with the reconstruction logic.

> you'll also need to be quite careful with custom space definitions. stoploss / roi / max_open_trades, (....) spaces can be [overridden](https://www.freqtrade.io/en/latest/advanced-hyperopt/#overriding-pre-defined-spaces) - where your assumptions won't hold true any longer (can be a point to add as warning to the docs just as well).

You're right. I mostly use a custom ROI search space myself, so yeah, I know what you're talking about. That must be documented.

File: freqtrade/optimize/hyperopt/hyperopt.py (hunk 2 above)
Comment (laloch) [reply freqtrade/optimize/hyperopt/hyperopt.py:233]: I was afraid someone is going to point that out ;)

**Comment by xmatthias** [comment]
@laloch are there plans to follow up / continue on this work? or was this a one-off drive-by contribution we're now supposed to take over/finish?
